src/Ideal.jl

```diff
@@ -4,15 +4,31 @@
 #
 ###############################################################################
 
-function ideal
+# We assume that the functions
+#   ideal(R::Ring, x::RingElement)
+#   ideal(R::Ring, xs::AbstractVector)
+# are implemented by anyone implementing ideals for AbstractAlgebra rings.
+# The functions in this file extend the interface for `ideal`.
```

Comment:
We really should add a new `docs/src/ideal_interface.md` file (with suitable content, modeled on the other `*_interface.md` files). But that's for a future PR, I'd say :-)

Reply:
Also a "conformance test" (like we have for groups, rings). Again: potential future work, not for this PR

Reply:
See issue #1733